cupy/_io/npz.py

```diff
@@ -1,17 +1,47 @@
+import collections.abc
 import warnings
 
 import numpy
 
 import cupy
 
-
 _support_allow_pickle = (numpy.lib.NumpyVersion(numpy.__version__) >= '1.10.0')
 
 
-class NpzFile(object):
+if numpy.lib.NumpyVersion(numpy.__version__) >= '2.0.0':
+    from numpy.lib._npyio_impl import BagObj
```

Comment:
Could you copy the `BagObj` class from NumPy? I think it's better not to rely on NumPy's implementation details.

Reply:
"Initially, I came up with the idea of having a separate BagObj class. However, the entire implementation tries to use NumPy's existing code, so I tried to follow the same approach. If you think it would be better to have a separate BagObj, I can implement it."